### PR TITLE
Shodan collector ciac 11405

### PR DESCRIPTION
**Pull request by YairGlik**
## Contributing to Cortex XSOAR Content
Make sure to register your contribution by filling the [contribution registration form](https://forms.gle/XDfxU4E61ZwEESSMA)

**The Pull Request will be reviewed only after the contribution registration form is filled.**

## Status
- [x] In Progress
- [ ] Ready
- [ ] In Hold - (Reason for hold)

## Related Issues
fixes: [CIAC-11405](https://jira-dc.paloaltonetworks.com/browse/CIAC-11405)

## Description
A few sentences describing the overall goals of the pull request's commits.

## Must have
- [ ] Tests
- [ ] Documentation

### CODE REVIEW

File: Packs/Shodan/Integrations/Shodan_v2/Shodan_v2.py [hunk 1]
```diff
@@ -446,6 +509,56 @@ def shodan_network_alert_remove_service_from_whitelist_command():
...
+    Returns:
+        tuple[Dict, List[Dict]]: A tuple where the first item is the updated last_run data,
+        and the second item is a list of fetched events.
+    """
+    events = http_request('GET', '/shodan/alert/info')
+    if not isinstance(events, list):
+        events = [events]
+    demisto.debug(f'Before filtering, {len(events)} events')
+
+    start_date = arg_to_datetime(last_run.get("start_date"))
```
Comment (jbabazadeh) [inline Packs/Shodan/Integrations/Shodan_v2/Shodan_v2.py:545]: i prefer to call it last_fetch_time or something like that.

File: Packs/Shodan/Integrations/Shodan_v2/Shodan_v2_description.md [hunk 2]
```diff
@@ -0,0 +1,10 @@
+Fetch process returns a listing of all the network alerts that are currently active on the account.
+
+To enable the Shodan integration you need to have an API key, which you can get for free by creating a Shodan account <https://account.shodan.io/register>
+Once you have an API key you insert it into the API Key field and click the Test button.
+ 
+Rate Limits:
+All API plans are subject to a rate limit of 1 request per second - [docs](https://account.shodan.io/billing)
+
+Integration description:
```
Comment (jbabazadeh) [inline Packs/Shodan/Integrations/Shodan_v2/Shodan_v2_description.md:9]: put this first and add titles  (remove the integration description)

**Review by jbabazadeh — CHANGES_REQUESTED** [review]
Great work!! please see my comments and please add more unit test to match the coverage that needed.

File: Packs/Shodan/Integrations/Shodan_v2/Shodan_v2.yml [hunk 3]
```diff
@@ -348,11 +357,33 @@ script:
...
+      description: If set to 'True', the command will create events; otherwise, it will only display them.
+      defaultValue: 'False'
+      auto: PREDEFINED
+      isArray: false
+      predefined:
+      - 'True'
+      - 'False'
+    - name: start_date
+      description: 'Fetch events created after this date. You can also use relative terms like "3 days ago".'
+      defaultValue: "3 days ago"
```
[the remark this answers is not in the document]
Comment (YairGlik) [reply Packs/Shodan/Integrations/Shodan_v2/Shodan_v2.yml:373]: According to the documentation - it brings me the list of all active alerts in the account,
And I filter it in the code

File: Packs/Shodan/Integrations/Shodan_v2/README.md [hunk 4]
```diff
@@ -1,4 +1,4 @@
-A search engine used for searching Internet-connected devices.
+Shodan is a search engine for Internet-connected devices. Unlike traditional search engines that index websites, Shodan indexes information about the devices connectedש to the internet, such as servers, routers, webcams, and other IoT devices.
```
Comment (jbabazadeh) [inline Packs/Shodan/Integrations/Shodan_v2/README.md:1]:
```suggestion
Shodan is a search engine for Internet-connected devices. Unlike traditional search engines that index websites, Shodan indexes information about the devices connected to the internet, such as servers, routers, webcams, and other IoT devices.
```

File: Packs/Shodan/Integrations/Shodan_v2/Shodan_v2.py [hunk 5]
```diff
@@ -174,11 +177,16 @@ def test_module():
...
     """
-    http_request('GET', '/shodan/ports', {'query': 'test'})
-
     params = demisto.params()
-    if argToBoolean(params.get('isFetchEvents', False)):
-        get_events_command({'max_fetch': 1, 'start_date': 'now'})
+    is_fetch_events = argToBoolean(params.get('isFetchEvents', False))
+
+    if is_fetch_events and not API_KEY:
+        return_error("Missing API key")
```
Comment (jbabazadeh) [inline Packs/Shodan/Integrations/Shodan_v2/Shodan_v2.py:184]:
```suggestion
        return_error("Missing API key - You must provide API KEY parameter.")
```

**Review by jbabazadeh — APPROVED** [review]
Great! just small comments. pleas ask doc review from @ShirleyDenkberg

File: Packs/Shodan/Integrations/Shodan_v2/README.md [hunk 6]
```diff
@@ -1,17 +1,19 @@
...
-    | **Parameter** | **Required** |
-    | --- | --- |
-    | Api Key | True |
-    | Base url to Shodan API | True |
-    | Trust any certificate (not secure) | False |
-    | Use system proxy settings | False |
+    | **Parameter** | **Description** | **Required** |
+    | --- | --- | --- |
+    | API Key |  | False |
+    | Base url to Shodan API |  | True |
```
Comment (ShirleyDenkberg) [inline Packs/Shodan/Integrations/Shodan_v2/README.md:12]:
```suggestion
    | Base URL to Shodan API |  | True |
```

File: Packs/Shodan/Integrations/Shodan_v2/README.md [hunk 7]
```diff
@@ -752,3 +754,36 @@ There is no context output for this command.
...
+
+There is no context output for this command.
+
+
+## Fetch Events
+
+Fetch process returns a listing of all the network alerts that are currently active on the account.
+
+To enable the Shodan integration you need to have an API key, which you can get for free by creating a Shodan account <https://account.shodan.io/register>
+Once you have an API key you insert it into the API Key field and click the Test button.
```
Comment (ShirleyDenkberg) [inline Packs/Shodan/Integrations/Shodan_v2/README.md:784]:
```suggestion
Once you have an API key, you insert it into the *API Key* field and click the **Test** button.
```

File: Packs/Shodan/Integrations/Shodan_v2/Shodan_v2_description.md [hunk 8]
```diff
@@ -0,0 +1,14 @@
+# Shodan v2
+
+Shodan is a search engine for Internet-connected devices. Unlike traditional search engines that index websites, Shodan indexes information about the devices connected to the internet, such as servers, routers, webcams, and other IoT devices.
+
+
+## Setup
+
+To enable the Shodan integration you need to have an API key, which you can get for free by creating a Shodan account <https://account.shodan.io/register>
+Once you have an API key you insert it into the API Key field and click the Test button.
```
Comment (ShirleyDenkberg) [inline Packs/Shodan/Integrations/Shodan_v2/Shodan_v2_description.md:9]:
```suggestion
Once you have an API key you insert it into the *API Key* field and click the **Test** button.
```

**Comment by ShirleyDenkberg** [comment]
@jbabazadeh Doc review completed.

**Comment by YairGlik** [comment]
Hi @DeanArbel,
There were no tests before, and I added tests only for the functions I wrote. As a result, the coverage is now below 70%. Please force merge.

**Pull request merged**